src/python/pants/backend/python/subsystems/python_native_code.py

```diff
@@ -47,10 +47,6 @@ def register_options(cls, register):
   def subsystem_dependencies(cls):
     return super(PythonNativeCode, cls).subsystem_dependencies() + (
       NativeToolchain.scoped(cls),
-      # We generally have to use PythonSetup's global instance, as methods such as
-      # `dump_requirements` will use it directly.
-      # TODO: when subsystems are more easily requestable from the v2 engine, this restriction could
-      # be removed!
       PythonSetup,
```

Comment:
Does removing this comment mean that this instance is no longer required either? Or is it still used in other spots?

Reply:
The instance is still used, that comment was introduced in 6b11c59ad9 to explain a switch from a scoped instance to the global instance. It seemed weird to me to have it in the code (as opposed to in the commit message for that change) because the casual reader doesn't know the history of the file and so this is addressing something they aren't wondering about to begin with. We could leave it in there as a warning against switching back to using the scoped instance, but that seems rather ad-hoc. We'd have to add that warning all over the place...